test_schools_ks.py

```diff
@@ -0,0 +1,111 @@
+        except FileNotFoundError:
+            print(f"Files for test instance: {self.name} does not exist", file=stderr)
+            pprint(self.__dict__)
+            exit(1)
+
+
+test_instances = [
+    TestInstance("aalborg", "in/cities/aalborg.net.xml", "in/cities/aalborg.stat.xml", "stats/aalborg.stat.xml"),
+    TestInstance("esbjerg", "in/cities/esbjerg.net.xml", "in/cities/esbjerg.stat.xml", "stats/esbjerg.stat.xml"),
+    TestInstance("randers", "in/cities/randers.net.xml", "in/cities/randers.stat.xml", "stats/randers.stat.xml"),
+    TestInstance("slagelse", "in/cities/slagelse.net.xml", "in/cities/slagelse.stat.xml", "stats/slagelse.stat.xml"),
+    TestInstance("vejen", "in/cities/vejen.net.xml", "in/cities/vejen.stat.xml", "stats/vejen.stat.xml")
+]
+
+
+def write_school_coords(net: sumolib.net.Net, stats: ET.ElementTree, filename):
+    """
+        Writes all schools' positions found in stats file to a csv called 'filename'
+        :param net: network file that the schools in stats file is placed on
+        :param stats: stats file parsed with ElementTree containing schools
+        :param filename: name of csv to be written
+    """
+    xml_schools = [xml_school for xml_school in stats.find("schools").findall("school")]
+
+    if xml_schools is None:
+        return
+
+    positions = []
+
+    # find all generated school positions, append to positions
+    for gen_school_edge in xml_schools:
+        pos = float(gen_school_edge.get("pos"))
+        edge = net.getEdge(gen_school_edge.get("edge"))
+        positions.append(position_on_edge(edge, pos))
+
+    # workaround to append columns to csv file. read old_csv, make a new csv and write to this, rename to old csv when done
+    old_csv = f'out/{filename}.csv'
+    new_csv = f'out/{filename}-new.csv'
+
+    # if the old_csv already exists, do as mentioned above
+    if os.path.exists(old_csv):
+        with open(old_csv, 'r') as read_obj, \
+                open(new_csv, 'w', newline='') as write_obj:
+            csv_reader = csv.reader(read_obj)
+            csv_writer = csv.writer(write_obj)
+            for i, row in enumerate(csv_reader):
+                row.append(positions[i][0])
+                row.append(positions[i][1])
+                csv_writer.writerow(row)
+        try:
+            os.rename(new_csv, old_csv)
+        except WindowsError:
+            os.remove(old_csv)
+            os.rename(new_csv, old_csv)
+    else:
+        # if file does not exist, simply write to it
+        file = open(old_csv, 'w', newline='')
+        with file:
+            writer = csv.writer(file)
+            writer.writerows(positions)
+
+
+def run_multiple_test(test: TestInstance, times: int):
+    # execute a test-instance n times
+    for n in range(0, times):
+        # find number of actual schools in the city
+        real_schools = len(
+            [xml_school for xml_school in ET.parse(test.real_stats_file).find("schools").findall("school")])
```

Comment:
This snippet can be moved out of the loop

Reply:
Renaming to `real_schools_count` is probably a good idea.

Reply:
ef820bf0aca82aa04c9d78eccae41601a48ffb40